networkdb/delegate.go

```diff
@@ -163,11 +163,27 @@ func (nDB *NetworkDB) handleNetworkEvent(nEvent *NetworkEvent) bool {
 	}
 
 	if n, ok := nodeNetworks[nEvent.NetworkID]; ok {
+		// If a leave event is received late network state shouldn't
+		// be updated, but cleanup the old table entries.
+		if nEvent.Type == NetworkEventTypeLeave {
```

Comment:
can this be moved under the lamport check? This way we don't trigger flush for every repeated leave message with old lamport